### PR TITLE
Remove malloc references in mbedtls/scripts

### PR DESCRIPTION
**Pull request by andresag01**
This change removes a few references to `malloc()` in the `scripts` folder. Recall that the use of `malloc()` has been replaced by `calloc()` across the library.

**NOTES:**
* As these changes are only in the scripts folder, I do not think it needs a ChangeLog entry.
* The changes needs to be backported to mbedtls 2.1, but not to mbedtls 1.3 as malloc is still used extensively there.

### CODE REVIEW

File: scripts/malloc-init.pl [hunk 1]
```diff
@@ -1,70 +0,0 @@
-#!/usr/bin/perl
```
Comment (RonEld) [inline scripts/malloc-init.pl]: why was this file deleted?

Comment (andresag01) [inline scripts/malloc-init.pl]: This script is trying to identify if we have malloc'ed something but did not initialise it with memset, memcpy,fread, etc. However, this is now a little redundant because we have migrated to calloc which initialises everything to 0 by default.

Comment (RonEld) [inline scripts/malloc-init.pl]: thanks, I see this file is not called anywhere, so It's ok to remove.
Have you checked th\at there aren't any references for this script anywhere? (articles, usage etc..)

Comment (andresag01) [inline scripts/malloc-init.pl]: Well, I am confident that the script is not used in any of our scripts, but I cannot say about our documentation. There are too many articles to check by hand. Do you think greping our documentation repository is sufficient?

Comment (RonEld) [inline scripts/malloc-init.pl]: Unfortunately our docs repository is not full yet, and a quick google search only showed the github links, so I think we are safe here

File: scripts/find-mem-leak.cocci [hunk 2]
```diff
@@ -2,8 +2,8 @@
 expression x, y;
 statement S;
 @@
-  x = mbedtls_malloc(...);
-  y = mbedtls_malloc(...);
+  x = mbedtls_calloc(...);
+  y = mbedtls_calloc(...);
   ...
```
Comment (hanno-arm) [inline scripts/find-mem-leak.cocci]: @andresag01 This question is not related to the switch from `calloc` to `malloc`, but how does `find-mem-leak.cocci` help detecting memory leaks?

Comment (andresag01) [inline scripts/find-mem-leak.cocci]: @hanno-arm: My understanding is that cocci is a sort of scripting language that is executed by `spatch`. This scripts essentially allow you to identify patterns and change the code. So in the case of the `find-mem-leak.cocci` script, we are trying to identify code of the form:
```c
x = mbedtls_calloc();
y = mbedtls_calloc();

if( x == NULL || y == NULL )
{
    return;
}
```
Now consider the case when one allocation succeeds and the other fails. Then either `x == NULL && y != NULL` or `x != NULL && y == NULL` and the `if()` statement will evaluate to true and return. This means that we are either leaking the memory allocating for y in the former case and x in the latter. This cocci script helps us flag some of these memory leaks.

Comment (hanno-arm) [inline scripts/find-mem-leak.cocci]: @andresag01 Thanks! I was wondering because this script would also flag valid, non-leaking code paths which call `free` on both `x` and `y` inside the check, and I think we're using this construct in several places in the library.

Comment (andresag01) [inline scripts/find-mem-leak.cocci]: @hanno-arm: Indeed, some of these scripts do flag false positives. As far as I can tell, we are not executing these scripts from any other test scripts.

Comment (hanno-arm) [inline scripts/find-mem-leak.cocci]: @andresag01 Ok. If supported by the semantic match language, one might restrict `S` to statements not involving `x` and `y` - but that's an issue independent from the switch to `calloc`.

Comment (andresag01) [inline scripts/find-mem-leak.cocci]: @hanno-arm: Agreed, I think improving the checks is a separate task and should be addressed in another PR.

**Comment by gilles-peskine-arm** [comment]
CI failed only due to timing tests (known issue), ok to merge
